region/brinstar/red/Red Tower.json

```diff
@@ -837,6 +837,43 @@
         ]}
       ]
     },
+    {
+      "link": [1, 9],
+      "name": "Shinespark (Come In Shinecharged)",
+      "entranceCondition": {
+        "comeInShinecharged": {
+          "framesRequired": 50
```

Comment:
sparking diagonally immediately on room entry can be done with a single frame. Probably worth modeling separately from the jump option ... but it may have different frames remaining, as youll be on the right side of the shaft ...

Reply:
True, added a strat for that now. Also realized that the framesRequired for the vertical spark could also be reduced by sparking closer to the door, though this breaks the platform so you have to land on the frozen enemies. I changed all the new strats to go directly to the door node (5) instead of the platform junction (9), as there doesn't appear to be anything useful you could do here starting from node 9 other than freezing the enemies and continuing to node 5 (e.g., if you could kill the enemies and use walljump, IBJ, or SpaceJump then there would be no need for the shinespark).